sonar-core/src/main/java/org/sonar/core/util/DefaultHttpDownloader.java

```diff
@@ -246,10 +272,17 @@ private void propagateProxySystemProperties(Map<String, String> settings) {
      * @since 5.0
      */
     public InputSupplier<InputStream> newInputSupplier(URI uri, String requestMethod, String login, String password, @Nullable Integer readTimeoutMillis) {
-      if (readTimeoutMillis != null) {
-        return new HttpInputSupplier(uri, requestMethod, userAgent, login, password, readTimeoutMillis);
-      }
-      return new HttpInputSupplier(uri, requestMethod, userAgent, login, password, TIMEOUT_MILLISECONDS);
+      return newInputSupplier(uri, requestMethod, login, password, null, readTimeoutMillis);
+    }
+
+    /**
+     * @since 5.2
+     */
+    public InputSupplier<InputStream> newInputSupplier(URI uri, String requestMethod, String login, String password, @Nullable Integer connectTimeoutMillis,
+      @Nullable Integer readTimeoutMillis) {
+      int readTimeout = readTimeoutMillis != null ? readTimeoutMillis : TIMEOUT_MILLISECONDS;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename "readTimeout" which hides the field declared at line 66. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AHiddenFieldCheck)